kubernetes/ansible/roles/logging/templates/oauth2-proxy.yaml

```diff
@@ -41,3 +50,20 @@ authenticatedEmailsFile:
      {% endfor %}
 
 fullnameOverride: "oauth2-proxy"
+
+{% if (graylog_open_to_public is not defined or graylog_open_to_public is false) and (send_logs_to_graylog is defined and send_logs_to_graylog is true) and (graylog_open_to_private is defined and graylog_open_to_private is true) %}
+service:
+  type: LoadBalancer
+  loadBalancerIP: {{ oauth2_proxy_lb_ip }}
+  port: 80
+{% if nginx_private_ingress_annotations is defined and nginx_private_ingress_annotations %}
+# you can define annotations in ansible for custom behaviour
+# eg:
+# nginx_private_ingress_annotations:
+#   cloud.google.com/load-balancer-type: "Internal"
+  annotations: {{nginx_private_ingress_annotations|to_json}}
+{% else %}
+  annotations:
+    service.beta.kubernetes.io/azure-load-balancer-internal: "true"
```

Comment:
Please remove this, as it'll only work in azure. All the other providers will ignore this and will create public LB. It's better to error out than an undesired behaviour.

Reply:
Done. Reusing nginx variable to not introduce another variable.